packages/contracts-bedrock/SOLIDITY_UPGRADES.md

```diff
@@ -0,0 +1,78 @@
+# Solidity Versioning Policy
+
+This document outlines the process for proposing and implementing Solidity version updates in the
+OP Stack codebase.
+
+## Unified Solidity Version
+
+The OP Stack codebase maintains a single, unified Solidity version across all contracts and
+components. This ensures consistency, simplifies maintenance, and reduces the risk of
+version-related issues.
+
+**Important**: New Solidity versions must not be introduced to any part of the codebase without
+going through the formal version update proposal process outlined in this document.
+
+## Update Process
+
+1. **Minimum Delay Period**: A new Solidity version must be at least 6 months old before it can be
+  considered for adoption.
+2. **Proposal Submission**: Before any Solidity version upgrades are made, a detailed proposal must
+  be submitted as a pull request to the [`ethereum-optimism/design-docs`](https://github.com/ethereum-optimism/design-docs)
+  repository in the `solidity/` subfolder, following the standardized format outlined below. This
+  applies to the entire codebase; individual components or contracts cannot be upgraded separately.
+3. **Review and Approval**: A dedicated review panel will assess the proposal based on the
+  following criteria:
+    - Is the Solidity version at least 6 months old?
+    - Does the proposed upgrade provide clear value to the codebase?
+    - Do any new features or bug fixes pose an unnecessary risk to the codebase?
+4. **Implementation**: If the proposal receives unanimous approval from the review panel, the
+  Solidity version upgrade will be implemented across the entire OP Stack codebase.
+
+## Proposal Submission Guidelines
+
+To submit a Solidity version upgrade proposal, create a new pull request to the `ethereum-optimism/design-docs` repository, adding a new file in the `solidity/` subfolder. Use the following format for the file name and content:
+
+### File Name
+
+```txt
+solc-update-[current-version]-to-[proposed-version].md
+```
+
+### File Content
+
+```markdown
+# Solidity Version Upgrade Proposal
+
+**Current Version:** [e.g., 0.8.20]
+**Proposed Version:** [e.g., 0.8.25]
+
+## Benefits to the OP Stack
+
+[Explain specific feature(s) in the new version that would benefit the OP Stack codebase, with a clear explanation of their potential impact and usefulness]
+
+## Summary of Changes
+
+[Provide a comprehensive summary of all significant features, changes, and bug fixes introduced between the current version and the proposed upgrade]
+
+## High-Risk Features or Notable Bug Fixes
+
+[Highlight any high-risk features or notable bug fixes that require special attention during the review process]
```

Comment:
What is a high risk feature?

Reply:
Yeah good point, I'm going to make that separate PR into design-docs first then we can get back to this

Reply:
https://github.com/ethereum-optimism/design-docs/pull/104